pkg/router/headerformatter.go

```diff
@@ -18,17 +18,36 @@
 package router
 
 import (
+	"context"
+	"mosn.io/mosn/pkg/variable"
+	"regexp"
 	"strings"
 
 	"mosn.io/mosn/pkg/log"
-	"mosn.io/mosn/pkg/types"
 )
 
+var variableValueRegexp = regexp.MustCompile(`^%.+%$`)
+
 func getHeaderFormatter(value string, append bool) headerFormatter {
-	// TODO: variable headers would be support very soon
+	if variableValueRegexp.MatchString(value) {
```

Comment:
maybe we don't need introduce regex, and it's easy to judge that first and last string are `%`.

Reply:
Benchmark shows that `strings.HasPrefix` and `strings.HasSuffix` are faster than regex......I will modify it later.
Another question is that if we got a value which not begins and ends with `%`, but contains '%', should we ignore it and return nil or treat it as a plain value?

Reply:
I think we should treat it as a plain value, beacause we do not prohibit `%` as a part of header